src/app/loadout-drawer/loadout-drawer-reducer.ts

```diff
@@ -523,10 +444,13 @@ export function fillLoadoutFromUnequipped(
 ): LoadoutUpdateFunction {
   return (loadout) => {
     const items = getUnequippedItemsForLoadout(store, category);
-    // TODO: batch addItems
     for (const item of items) {
-      // Add as an unequipped item
-      loadout = addItem(defs, item, false)(loadout);
+      // Don't mess with something that's already there
+      const dupeIndex = findSameLoadoutItemIndex(defs, loadout.items, item);
```

Comment:
This is a behavior change / bugfix - previously, existing equipped items could get unequipped if they were re-added by this function. Now they're left alone.